### ISSUE TITLE
extend `redefined-outer-name` for global vs class scope names

### ISSUE THREAD

**Issue by zufuliu** [pylint-dev/pylint#8677]
### Current problem

see https://discuss.python.org/t/pep-709-one-behavior-change-that-was-missed-in-the-pep/26691

```python
incr = 1
def func():
    incr = 2
    return [x+incr for x in range(3)]

class C:
    incr = 2
    items = [x+incr for x in range(3)]
```

pylint emits `W0621: Redefining name 'incr' from outer scope (line 1) (redefined-outer-name)` for `incr = 2` inside `func`, but not inside `class C`.

### Desired solution

emits `redefined-outer-name` for both places.

**Comment by Pierre-Sassoulas** [comment]
I'm not sure it's a false positive, ``incr`` is a class attribute in class C, i.e. will be accessed as ``c = C().incr`` so there's no namespace ambiguity. By contrast the following correctly raises because it's ambiguous once again:

```
class C:
    def d(self):
        incr = 2
        items = [x+incr for x in range(3)]
```    
What do you think ?

**Comment by jacobtylerwalls** [comment]
Duplicate of #5678. `redefined-outer-name` is pretty dumb, it only works in your first case (functions shadowing a global) or nested except handlers (lol).
